app/routes/upload.step3.tsx

```diff
@@ -1,5 +1,37 @@
+import { LoaderFunctionArgs } from '@remix-run/node'
+import { useLoaderData } from '@remix-run/react'
+import { formCookie } from '~/lib/cookies'
+
+export const loader = async ({ request }: LoaderFunctionArgs) => {
+  const cookieHeader = request.headers.get('Cookie')
+  const cookieData = await formCookie.parse(cookieHeader)
+
+  return { cookieData }
+}
+
 const step3 = () => {
```

Comment:
**issue**: Avoid using `@ts-ignore` as it suppresses TypeScript errors. Instead, address the underlying issue causing the error.